### PR TITLE
add testcase in generic_error_test.go

### PR DESCRIPTION
**Pull request by chchliang**
add testcase with function `Error(),Code()`  in generic_error_test.go 

Signed-off-by: chchliang <chen.chuanliang@zte.com.cn>

### CODE REVIEW

**Comment by crosbymichael** [comment]
LGTM

File: libcontainer/generic_error_test.go [hunk 1]
```diff
@@ -12,3 +12,39 @@ func TestErrorDetail(t *testing.T) {
...
+
+	for _, v := range cc {
+		err := newSystemErrorWithCause(fmt.Errorf(v.errmsg), v.cause)
+		if v.cause == "" {
+			if msg := err.Error(); msg != v.errmsg {
+				t.Fatalf("expected err msg %q but %q", v.errmsg, msg)
+			}
+		} else {
+			if msg := err.Error(); msg == v.errmsg {
+				t.Fatalf("expected err msg %q but %q", msg, v.errmsg)
```
Comment (mrunalp) [inline libcontainer/generic_error_test.go]: Here we are testing that they aren't equal but the message doesn't say that.

Comment (chchliang) [inline libcontainer/generic_error_test.go]: @mrunalp 
I haved changed the error messages.

**Comment by dqminh** [comment]
LGTM

File: libcontainer/generic_error_test.go [hunk 2]
```diff
@@ -12,3 +12,38 @@ func TestErrorDetail(t *testing.T) {
...
+
+	for _, v := range cc {
+		err := newSystemErrorWithCause(fmt.Errorf(v.errmsg), v.cause)
+
+		msg := err.Error()
+		if v.cause == "" && msg != v.errmsg {
+			t.Fatalf("expected err(%q) equal errmsg(%q)", msg, v.errmsg)
+		}
+		if v.cause != "" && msg == v.errmsg {
+			t.Fatalf("upexpected err(%q) equal errmsg(%q)", msg, v.errmsg)
```
Comment (mrunalp) [inline libcontainer/generic_error_test.go]: typo: upexpected -> unexpected

Comment (chchliang) [inline libcontainer/generic_error_test.go]: @mrunalp 
I have modified. thanks!

**Comment by cyphar** [comment]
LGTM.

**Comment by dqminh** [comment]
LGTM

**Pull request merged**